src/EnergyPlus/SolarShading.cc

```diff
@@ -4592,6 +4686,22 @@ namespace SolarShading {
 						}
 					}
 				}
+				// Elimate shawdowing surfaces that is supposed to be disabled.
+				if ( DisableAllSelfShading ) {					
+					if ( surface.Zone != 0 ) {
+						continue; // Disable all shadowing surfaces in all zones. Attached shading surfaces are not part of a zone, zone value is 0.
+					}
+				} else if ( DisableGroupSelfShading ) {
+					std::vector< int > DisabledZones = Surface( CurSurf ).DisabledShadowingZoneList;
+					bool isDisabledShadowSurf = false;
+					for ( int i : DisabledZones ) {
+						if ( surface.Zone == i ) {
+							isDisabledShadowSurf = true;
+							break;
+						}
+					}
+					if ( isDisabledShadowSurf ) continue; // Disable all shadowing surfaces in all disabled zones.
```

Comment:
And so calling `continue` here will start the next loop.  Which will skip applying shading to this surface.  Right?  Is that the intent?  Do you have any results showing these code changes were successful?

Reply:
@Myoldmopar for the current surface that calls this SHDGSS function, while looping through all its shading surfaces, skip (continue) the shading surfaces included in its disabled list. I have a unit test to check which should be included in the disabled list for all exteriror surfaces.